src/models/service/transactions/details.rs

```diff
@@ -155,41 +155,43 @@ pub struct MultisigExecutionDetails {
     pub safe_tx_gas: usize,
     pub base_gas: usize,
     pub gas_price: String,
+    // As this is a token we will keep the token information separated
     pub gas_token: String,
-    pub refund_receiver: String,
+    pub refund_receiver: AddressEx,
     pub safe_tx_hash: String,
-    pub executor: Option<String>,
-    pub signers: Vec<String>,
+    pub executor: Option<AddressEx>,
+    pub signers: Vec<AddressEx>,
     pub confirmations_required: u64,
     pub confirmations: Vec<MultisigConfirmation>,
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub rejectors: Option<Vec<String>>,
+    pub rejectors: Option<Vec<AddressEx>>,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub gas_token_info: Option<TokenInfo>,
 }
 
 #[derive(Serialize, Debug, PartialEq)]
 #[serde(rename_all = "camelCase")]
 pub struct MultisigConfirmation {
-    pub signer: String,
+    pub signer: AddressEx,
     pub signature: Option<String>,
     pub submitted_at: i64,
 }
 
 #[derive(Serialize, Debug, PartialEq)]
 #[serde(rename_all = "camelCase")]
 pub struct ModuleExecutionDetails {
-    pub address: String,
+    pub address: AddressEx,
 }
 
 #[derive(Serialize, Debug, PartialEq)]
 #[serde(rename_all = "camelCase")]
 pub struct TransactionData {
     pub hex_data: Option<String>,
     pub data_decoded: Option<DataDecoded>,
-    pub to: String,
+    pub to: AddressEx,
     pub value: Option<String>,
     pub operation: Operation,
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub address_info_index: Option<HashMap<String, AddressInfo>>,
+    // Mapping with info for the addresses in data_decoded
+    pub address_info_index: Option<HashMap<String, AddressEx>>,
```

Comment:
Like you mention before, as we have it, we avoid duplicates, but we could implement it in a way to avoid duplicates while still service `Vec<AddressEx>` in a separate ticket.

Reply:
But returning a dictionary to the clients will give the clients the guarantee that each address has one AddressEx. And they can easily look it up without having to build the map themselves beforehand. I don'T think we win much by avoiding 1 duplicated address. We do not have to optimize the response on every char.